modules/getComponents.js

```diff
@@ -11,35 +9,9 @@ function getComponentsForRoute(nextState, route, callback) {
   const getComponent = route.getComponent || route.getComponents
   if (!getComponent) {
     callback()
-    return
-  }
-
-  const { location } = nextState
-  let nextStateWithLocation
-
-  if (__DEV__ && canUseMembrane) {
-    nextStateWithLocation = { ...nextState }
-
-    // I don't use deprecateObjectProperties here because I want to keep the
-    // same code path between development and production, in that we just
-    // assign extra properties to the copy of the state object in both cases.
-    for (const prop in location) {
-      if (!Object.prototype.hasOwnProperty.call(location, prop)) {
-        continue
-      }
-
-      Object.defineProperty(nextStateWithLocation, prop, {
-        get() {
-          warning(false, 'Accessing location properties from the first argument to `getComponent` and `getComponents` is deprecated. That argument is now the router state (`nextState`) rather than the location. To access the location, use `nextState.location`.')
-          return location[prop]
-        }
-      })
-    }
   } else {
-    nextStateWithLocation = { ...nextState, ...location }
+    getComponent.call(route, nextState, callback)
```

Comment:
Oh, I see what you did – the idea with the `!getComponent` check above and the reason I just bail out there is because it's intended as a guard clause. Otherwise we should probably invert the order of the branches.